inspectit-oce-bootstrap/src/main/java/rocks/inspectit/oce/bootstrap/instrumentation/IGenericDataProvider.java

```diff
@@ -0,0 +1,21 @@
+package rocks.inspectit.oce.bootstrap.instrumentation;
+
+/**
+ * Interface implemented by all Data-Providers which are generated by the DataProviderGenerator.
+ * The generated class use the template GenericDataProviderTemplate.
+ */
+public interface IGenericDataProvider {
+
+    /**
+     * This interface is implemented by all DataProviders which are generated via CodeInjection.
+     *
+     * @param instrumentedMethodArgs the arguments passed to the method for which the data provider is executed
+     * @param thiz                   the "this" instance of the invoked method, null if the invoked method is static
+     * @param returnValue            the return value returned by the target method, if this data provider is executed at the end and no exception was thrown
+     * @param thrown                 the exception thrown bz the instrumented method, null otherwise
+     * @param additionalArgs         additional arguments, such as constants or values from the data cloud which are passed to this provider
+     * @return the collected datum or null if nothing was collected
+     */
+    Object execute(Object[] instrumentedMethodArgs, Object thiz, Object returnValue, Throwable thrown, Object[] additionalArgs);
```

Comment:
should we separate into two for pre and post collection? otherwise, you always need to check for non-null values in `returnValue` and `thrown`